.github/workflows/ci.yml

```diff
@@ -133,7 +133,20 @@ jobs:
           key: ${{ github.job }}-${{ matrix.name }}
           save: ${{ needs.setup.outputs.write-caches == 1 }}
       - name: CMake - configure
+        env:
+          JOB_NAME: ${{matrix.name}}
         run: |
+          if [[ "$JOB_NAME" == *"ubuntu"* ]]; then
+            DRIVER_OPTIONS="-DIREE_HAL_DRIVER_CUDA=ON -DIREE_HAL_DRIVER_HIP=ON"
+          elif [[ "$JOB_NAME" == *"macos"* ]]; then
+            DRIVER_OPTIONS="-DIREE_HAL_DRIVER_METAL=ON -DIREE_HAL_DRIVER_VULKAN=OFF"
+          elif [[ "$JOB_NAME" == *"windows"* ]]; then
+            DRIVER_OPTIONS="-DIREE_HAL_DRIVER_CUDA=ON -DIREE_HAL_DRIVER_HIP=ON"
+          else
+          # Disable CUDA and HIP on an unknow platform
+            DRIVER_OPTIONS="-DIREE_HAL_DRIVER_CUDA=OFF -DIREE_HAL_DRIVER_HIP=OFF"
+          fi
```

Comment:
You can fold these into the `matrix`:

```yml
      matrix:
        include:
          - name: ubuntu-20.04
            runs-on: ubuntu-20.04
            driver-options: "-DIREE_HAL_DRIVER_CUDA=ON -DIREE_HAL_DRIVER_HIP=ON"
          - name: windows-2022
            runs-on: windows-2022
            driver-options: "-DIREE_HAL_DRIVER_CUDA=ON -DIREE_HAL_DRIVER_HIP=ON"
          - name: macos-14
            runs-on: macos-14
            driver-options: "-DIREE_HAL_DRIVER_METAL=ON -DIREE_HAL_DRIVER_VULKAN=OFF"
```

alternately, that could set a CMake preset like

```yml
      matrix:
        include:
          - name: ubuntu-20.04-all-drivers
            runs-on: ubuntu-20.04
            cmake-preset: linux-all-drivers
          - name: ubuntu-20.04-minimal
            runs-on: ubuntu-20.04
            cmake-preset: linux-minimal
```

I'd go with the incremental step in this PR (my 1st suggestion), but I think the presets could make these workflow files more concise + easier to reproduce in developer workflows